File: src/worker/core/season/getSchedule.rb
async def get_schedule(one_day = false)
  schedule = await idb.cache.schedule.get_all

  return schedule if schedule[0].nil?

  if one_day
    partial_schedule = []
    team_ids = Set.new
    schedule.each do |game|
      break if game.day != schedule[0].day # Only keep games from same day
      break if team_ids.include?(game.home_tid) || team_ids.include?(game.away_tid) # Only keep games from unique teams, no 2 games in 1 day

      # For ASG and trade deadline, make absolutely sure they are alone. This shouldn't be necessary because add_days_to_schedule should handle it, but just in case...
      break if (game.home_tid < 0 || game.away_tid < 0) && team_ids.size > 0

      partial_schedule.push(game)
      team_ids.add(game.home_tid)
      team_ids.add(game.away_tid)

      # For ASG and trade deadline, make absolutely sure they are alone. This shouldn't be necessary because add_days_to_schedule should handle it, but just in case...
      break if game.home_tid < 0 || game.away_tid < 0
    end
    return partial_schedule
  end

  return schedule
end

export_default get_schedule

File: src/worker/core/season/getSchedule.cpp
#include <vector>
#include <set>
#include <optional>
#include "db/index.h" // Assuming a corresponding C++ DB header file
#include "common/types.h" // Assuming a corresponding C++ types header file

/**
 * Get an array of games from the schedule.
 *
 * @param {bool} oneDay Return just one day (true) or all days (false). Default false.
 * @return {std::vector<ScheduleGame>} Resolves to the requested schedule array.
 */
std::vector<ScheduleGame> getSchedule(bool oneDay = false) {
    std::vector<ScheduleGame> schedule = idb::cache::schedule::getAll();

    if (schedule.empty()) {
        return schedule;
    }

    if (oneDay) {
        std::vector<ScheduleGame> partialSchedule;
        std::set<int> teamIds;

        for (const auto& game : schedule) {
            if (game.day != schedule[0].day) {
                // Only keep games from same day
                break;
            }
            if (teamIds.count(game.homeTid) || teamIds.count(game.awayTid)) {
                // Only keep games from unique teams, no 2 games in 1 day
                break;
            }

            // For ASG and trade deadline, make absolutely sure they are alone. This shouldn't be necessary because addDaysToSchedule should handle it, but just in case...
            if ((game.homeTid < 0 || game.awayTid < 0) && !teamIds.empty()) {
                break;
            }

            partialSchedule.push_back(game);
            teamIds.insert(game.homeTid);
            teamIds.insert(game.awayTid);

            // For ASG and trade deadline, make absolutely sure they are alone. This shouldn't be necessary because addDaysToSchedule should handle it, but just in case...
            if (game.homeTid < 0 || game.awayTid < 0) {
                break;
            }
        }
        return partialSchedule;
    }

    return schedule;
}
